core/src/androidTest/java/com/pierfrancescosoffritti/androidyoutubeplayer/core/YoutubePlayerViewDefaultWebUITest.kt

```diff
@@ -0,0 +1,170 @@
+                ),
+                withClassName(
+                        containsString(
+                                FrameLayout::class.java.canonicalName)
+                ),
+                withParent(
+                        withClassName(
+                                containsString(
+                                        LegacyYouTubePlayerView::class.java.canonicalName)
+                        )
+                )
+        )
+    }
+
+
+    @Test
+    fun testIfDefaultIsWebUi() {
+        // Prepare
+
+        // Act
+
+        // Assert
+        onView(
+                getNativeUiViewGroup(R.id.youtube_player_view)
+        ).check(matches(hasChildCount(0)))
+    }
+
+    @Test
+    fun testIfNativeViewIsSetByXMLAttribute() {
+        // Prepare
+
+        // Act
+
+        // Assert
+        onView(
+                getNativeUiViewGroup(R.id.youtube_player_view_native)
+        ).check(
+                matches(
+                        not(hasChildCount(0))
+                )
+        )
+    }
+
+    @Test
+    fun testIfDefaultWebUiInitializedProgrammaticallyUsingInitializer1() {
+        // Prepare
+
+        // Act
+        testActivity.runOnUiThread {
+            youTubePlayerViewUsingInitializer1.initialize(object : AbstractYouTubePlayerListener() {
+
+            })
+        }
+
+        // Assert
+        onView(
+                getNativeUiViewGroup(R.id.youtube_player_view_manual_initialize_1)
+        ).check(matches(hasChildCount(0)))
+    }
+
+    @Test
+    fun testIfDefaultWebUiInitializedProgrammaticallyUsingInitializer2() {
+        // Prepare
+
+        // Act
+        testActivity.runOnUiThread {
+            youTubePlayerViewUsingInitializer2.initialize(object : AbstractYouTubePlayerListener() {
+
+            }, true)
+        }
+
+        // Assert
+        onView(getNativeUiViewGroup(R.id.youtube_player_view_manual_initialize_2)
+        ).check(matches(hasChildCount(0)))
+    }
+
+    @Test
+    fun testIfDefaultWebUiInitializedProgrammaticallyUsingInitializer3() {
+        // Prepare
+
+        // Act
+        testActivity.runOnUiThread {
+            youTubePlayerViewUsingInitializer3.initialize(object : AbstractYouTubePlayerListener() {
+
+            }, true, IFramePlayerOptions.Builder().controls(1).build())
+        }
+
+        // Assert
+        onView(
+                getNativeUiViewGroup(R.id.youtube_player_view_manual_initialize_3)
+        ).check(matches(hasChildCount(0)))
+    }
+
+    @Test
+    fun testIfNativeUiInitializedProgrammatically() {
+        // Prepare
+        youTubePlayerViewUsingInitializeWithNativeUi.initializeWithNativeUi(object : AbstractYouTubePlayerListener() {
+
+        }, true)
+        // Act
+
+        // Assert
+        onView(
+                getNativeUiViewGroup(R.id.youtube_player_view_manual_initialize_native)
+        ).check(
+                matches(
+                        not(hasChildCount(0))
+                )
+        )
+    }
+}
```

Comment:
We should also test the case in which native ui is enable.

Reply:
I think the second test does this?

Reply:
Ah ok, my bad.